test/blackbox-tests/test-cases/alias-module.t

```diff
@@ -16,3 +16,5 @@ Test alias module for wrapped libraries:
   
   (** @canonical Foo.Bar *)
   module Bar = Foo__Bar
+  
```

Comment:
I'm not super familiar with how wrapping is implemented. Here's how I understand it, with some questions:
- `Foo__` contains aliases to the various modules in the wrapped library so that `Foo__A` can be referred to as `A` when compiling `Foo__B`.
- question: is `Foo__` installed?
- It is possible to refer to `Foo__` which is sort of equivalent to `Foo`, but since modules are built with `-open Foo__`, this "empty" module `Foo__` is found instead
- question: does it catch this mistake in external code (that uses library `foo`)?
- question: there's already a check like "you're refering to module X which is being define, this does not make sense". does this replace this check?

Reply:
> Foo__ contains aliases to the various modules in the wrapped library so that Foo__A can be referred to as A when compiling Foo__B.

Indeed.

> question: is Foo__ installed?

Yes. `foo__.cmi` must be available

> It is possible to refer to Foo__ which is sort of equivalent to Foo, but since modules are built with -open Foo__, this "empty" module Foo__ is found instead

Right

> question: does it catch this mistake in external code (that uses library foo)?

Nope, but it can be modified to do so. Although it will cost a bit of overhead.

> question: there's already a check like "you're refering to module X which is being define, this does not make sense". does this replace this check?

That's when you refer to module `Lib` from `Lib`. The use case we're disallowing here is legal and currently allowed, but we'd like to restrict it because it relies on undocumented details.

Reply:
ok, thanks! the limitation seems good enough